src/appmixer/google/gmail/gmail-commons.js

```diff
@@ -0,0 +1,23 @@
+'use strict';
+
+const BASE_URL = 'https://gmail.googleapis.com/gmail/v1';
+
+module.exports = {
+    async fetchData(context, endpoint, options = {}) {
```

Comment:
what about

```js
       async fetchData(context, endpoint, {
        method = 'GET',
        params = {},
        headers = {}
    } = {}) {

        const options = {
            method,
            url: `${BASE_URL}${endpoint}`,
            headers: {
                Authorization: `Bearer ${context.auth.accessToken}`,
                ...headers
            },
            params
        };

        return await context.httpRequest(options);
    }

```
 - it's clear what is expected in the `options` object
 - saved some space, I found some variable redundant 
 - now you don't need to send any `options` from the `ListSignatures`